### PR TITLE
[SYCL] Allow ESIMD backend only when explicitly set by ONEAPI_DEVICE_SELECTOR

### PR DESCRIPTION
**Pull request by raaiq1**
ONEAPI_DEVICE_SELECTOR would previously implicitly allow ESIMD backends . This isn't desired behavior; desired behavior is to only allow ESIMD backend only when explicitly specified by ONEAPI_DEVICE_SELECTOR.
Fix suggested by @lbushi25 
Signed-off-by: Rauf, Rana <rana.rauf@intel.com>

### CODE REVIEW

File: sycl/source/detail/device_filter.cpp [hunk 1]
```diff
@@ -237,10 +237,12 @@ ods_target_list::ods_target_list(const std::string &envStr) {
 // 1. Filter backend is '*' which means ANY backend.
 // 2. Filter backend match exactly with the given 'Backend'
 bool ods_target_list::backendCompatible(backend Backend) {
+  
+  bool isESIMD= Backend == backend::ext_intel_esimd_emulator;
   return std::any_of(
       TargetList.begin(), TargetList.end(), [&](ods_target &Target) {
         backend TargetBackend = Target.Backend.value_or(backend::all);
-        return (TargetBackend == Backend) || (TargetBackend == backend::all);
+        return isESIMD ? (TargetBackend == Backend) : (TargetBackend == Backend) || (TargetBackend == backend::all);
```
Comment (steffenlarsen) [inline sycl/source/detail/device_filter.cpp:245]:
```suggestion
        return (TargetBackend == Backend) || (TargetBackend == backend::all) && !isESIMD;
```
You can also add the parenthesis if you'd like.

**Review by steffenlarsen — CHANGES_REQUESTED** [review]
Formatting doesn't seem to have been applied but lint hasn't complained, which is a little concerning. @raaiq1 - Could you please format the changes?

@pvchupin - Is this a know issue?

File: sycl/source/detail/device_filter.cpp (hunk 1 above)
Comment (raaiq1) [reply sycl/source/detail/device_filter.cpp]: Changes applied

**Review by steffenlarsen — APPROVED** [review]
LGTM! Thanks!

**Comment by pvchupin** [comment]
> @pvchupin - Is this a know issue?

I'm not aware about this issue.

**Pull request merged**